bundle/config/mutator/populate_current_user.go

```diff
@@ -25,13 +39,32 @@ func (m *populateCurrentUser) Apply(ctx context.Context, b *bundle.Bundle) diag.
 	if b.Config.Workspace.CurrentUser != nil {
 		return nil
 	}
-
+	m.initializeCache(ctx, b)
 	w := b.WorkspaceClient()
-	me, err := w.CurrentUser.Me(ctx)
+
+	var me *iam.User
+	var err error
+
+	fingerprint := b.GetUserFingerprint(ctx)
+	if !fingerprint.IsEmpty() {
+		log.Debugf(ctx, "[Local Cache] local cache is enabled")
+		me, err = m.cache.GetOrCompute(ctx, fingerprint, func(ctx context.Context) (*iam.User, error) {
+			currentUser, err := w.CurrentUser.Me(ctx)
+			return currentUser, err
+		})
+	} else {
+		log.Debugf(ctx, "[Local Cache] local cache is disabled")
+		me, err = w.CurrentUser.Me(ctx)
```

Comment:
How can the fingerprint be empty?

If we make sure `m.cache` is always valid, and will always return, do we need this branch?